homeassistant/components/myuplink/sensor.py

```diff
@@ -0,0 +1,161 @@
+"""Sensor for myUplink."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from myuplink.models import DevicePoint
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import MyUplinkDataCoordinator
+from .const import DOMAIN
+from .coordinator import CoordinatorData
+from .entity import MyUplinkEntity
+
+
+@dataclass(kw_only=True)
+class MyUplinkDeviceSensorEntityDescription(SensorEntityDescription):
+    """Describes MyUplink device sensor entity."""
+
+    value_fn: Callable[[CoordinatorData, str], StateType]
+
+
+DEVICE_SENSORS: tuple[MyUplinkDeviceSensorEntityDescription, ...] = (
+    MyUplinkDeviceSensorEntityDescription(
+        key="firmware_current",
+        name="Firmware Current",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data, device_id: data.devices[device_id].firmwareCurrent,
+    ),
+    MyUplinkDeviceSensorEntityDescription(
+        key="firmware_desired",
+        name="Firmware Desired",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data, device_id: data.devices[device_id].firmwareDesired,
+    ),
+    MyUplinkDeviceSensorEntityDescription(
+        key="connection_state",
+        name="Connection state",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data, device_id: data.devices[device_id].connectionState,
+    ),
+)
+
+DEVICE_POINT_DESCRIPTIONS = {
+    "°C": SensorEntityDescription(
```

Comment:
I added one description for the Celsius device point as that case was already existing. Please extend the device point descriptions with more measurement types in follow up PRs.